PrestaShop/PrestaShop
# FO: Contextual notifications

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Displays notifications that can be disabled by the user for the current shop or group context (by closing it), using the localstorage. 
| Type?             | new feature
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #23739
| How to test?      | see issue #23739 and description below
| Possible impacts? | Nothing special

# How to test

- Activate multistore
- Create a new shop group, do not set a color
- Create a new shop, do not set a color
- Configure new shop url
- Now go on maintenance page (because it has multistore header + multistore checkboxes)
- test your new shop and group contexts, see if messages appear and behave like described in the issue

To reset your test, you can remove the `contextual_notifications` in your localstorage, in the browser console.

# Technical description

This PR makes it possible to display a notification conditionally, the information (should be displayed or not?) is stored in localstorage. For this I created a new `contextual-notification` js component, responsible for initiating the localstorage config object if it doesn't exist, then setting or getting the data for a given config key, it also has a method to display the wanted notification.




---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/24185)
